src/lib/indexer/index.ts

```diff
@@ -254,4 +277,59 @@ export default class Indexer {
       }
     })
   }
+  getStatus = async (): Promise<Status> => {
+    const indexerTimeout = this.env.INDEXER_TIMEOUT_MS
+    const currentDate = new Date()
+    if (currentDate.getTime() - this.startupTime.getTime() < indexerTimeout) {
+      // if we started less than 30s ago -> PASS
+      return {
+        status: serviceState.UP,
+        detail: {
+          message: 'Service healthy. Starting up.',
+          startupTime: this.startupTime,
+          latestActivityTime: currentDate,
+        },
+      }
+    }
+    if (this.lastProcessedBlockTime === null && this.lastUnprocessedBlockTime === null) {
+      return {
+        status: serviceState.DOWN,
+        detail: {
+          message: 'Last activity was more than 30s ago, no blocks were processed.',
+          startupTime: this.startupTime,
+          latestActivityTime: null,
+        },
+      }
+    }
+    const latestActivityTime = this.lastProcessedBlockTime || this.lastUnprocessedBlockTime
+    if (latestActivityTime === null) throw new Error('sth')
```

Comment:
you can remove this `if` by defining `latestActivityTime` before the `this.lastProcessedBlockTime === null && this.lastUnprocessedBlockTime === null` check. The equivalent is then if `latestActivityTime === null`

Reply:
I've been more explicit in further comments below